src/App.tsx

```diff
@@ -1,51 +1,110 @@
 import './App.scss';
+import './Goods.scss';
 
-// import goodsFromServer from './goods';
-
-export const App: React.FC = () => {
-  return (
-    <main className="App">
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
-
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
-
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
-
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
-
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
+import goodsFromServer from './goods';
+
+type State = {
+  selectedGoods: string[];
 };
+
+export class App extends Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  clearSelected = () => this.setState({ selectedGoods: [] });
+
+  removeGood = (selectedGood: string) => {
+    this.setState((state) => (
+      {
+        selectedGoods: state.selectedGoods
+          .filter(good => good !== selectedGood),
+      }
+    ));
+  };
+
+  addGood = (selectedGood: string) => {
+    this.setState((prevState) => (
+      { selectedGoods: [...prevState.selectedGoods, selectedGood] }
+    ));
+  };
+
+  formatTitle = (selectedGoods: string[]) => {
+    const selectedCount = selectedGoods.length;
+
+    switch (selectedCount) {
+      case 0:
+        return 'No goods selected';
+      case 1:
+        return `${selectedGoods} is selected`;
+      default:
+        return `${selectedGoods.slice(0, -1).join(', ')} and ${selectedGoods[selectedCount - 1]} are selected`;
+    }
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <main className="App panel is-primary">
+        <header className="App__header panel-heading">
+          <h1 className="App__title">
+            {this.formatTitle(selectedGoods)}
+          </h1>
+
+          {selectedGoods.length
+            ? (
+              <button
+                type="button"
+                className="App__clear button is-danger"
+                onClick={this.clearSelected}
+              >
+                Clear
+              </button>
+            ) : ''}
+        </header>
+
+        <ul>
+          {goodsFromServer.map(good => {
+            const isActive = selectedGoods.includes(good);
+
+            return (
+              <li
+                key={uuid()}
```

Comment:
No, it's a bad usage
Read this thread - https://twitter.com/dan_abramov/status/1415279090446204929

Reply:
> No, it's a bad usage Read this thread - https://twitter.com/dan_abramov/status/1415279090446204929

My bad🤦‍♂️